apps/microtvm/arduino/template_project/microtvm_api_server.py

```diff
@@ -0,0 +1,378 @@
+        # directories in the process. Also copies this file, microtvm_api_server.py,
+        # in case TVM needs to call it from the new location
+        if IS_TEMPLATE:
+            shutil.copytree(API_SERVER_DIR, project_dir, dirs_exist_ok=True)
+
+        # Reference key directories with pathlib
+        project_dir = pathlib.Path(project_dir)
+        source_dir = project_dir / "src"
+
+        # Copy standalone_crt into src folder
+        self._copy_standalone_crt(source_dir, standalone_crt_dir)
+        self._remove_unused_components(source_dir)
+        self._remove_unit_tests(project_dir)
+
+        # Unpack the MLF and copy the relevant files
+        graph = self._disassemble_mlf(model_library_format_path, source_dir)
+
+        shutil.copy2(model_library_format_path, source_dir / "model")
+
+        # Populate our parameters file
+        self._populate_parameters_file(graph, source_dir)
+
+        # Recursively change imports
+        self._convert_imports(project_dir, source_dir)
+
+    def _get_fqbn(self, options):
+        o = BOARD_PROPERTIES[options["arduino_board"]]
+        return f"{o['package']}:{o['architecture']}:{o['board']}"
+
+    def build(self, options):
+        BUILD_DIR.mkdir()
+        print(BUILD_DIR)
+
+        compile_cmd = [
+            options["arduino_cmd"],
+            "compile",
+            "./project/",
+            "--fqbn",
+            self._get_fqbn(options),
+            "--build-path",
+            BUILD_DIR.resolve(),
+        ]
+
+        if options.get("verbose"):
+            compile_cmd.append("--verbose")
+
+        # Specify project to compile
+        output = subprocess.check_call(compile_cmd)
+        assert output == 0
+
+    # We run the command `arduino-cli board list`, which produces
+    # outputs of the form:
+    """
+    Port         Type              Board Name FQBN                          Core
+    /dev/ttyS4   Serial Port       Unknown
+    /dev/ttyUSB0 Serial Port (USB) Spresense  SPRESENSE:spresense:spresense SPRESENSE:spresense
+    """
+
+    def _auto_detect_port(self, options):
+        list_cmd = [options["arduino_cmd"], "board", "list"]
+        list_cmd_output = subprocess.check_output(list_cmd).decode("utf-8")
+        # Remove header and new lines at bottom
+        port_options = list_cmd_output.split("\n")[1:-2]
+
+        # Select the first compatible board
+        fqbn = self._get_fqbn(options)
+        for port_option in port_options:
+            if fqbn in port_option:
+                return port_option.split(" ")[0]
+
+        # If no compatible boards, raise an error
+        raise BoardAutodetectFailed
+
+    def _get_arduino_port(self, options):
+        if not self._port:
+            if "port" in options and options["port"]:
+                self._port = options["port"]
+            else:
+                self._port = self._auto_detect_port(options)
+
+        return self._port
+
+    def _get_baudrate(self, options):
```

Comment:
here you could probably skip the function since you're just hardcoding it for now